_posts/2017-12-13-Docker-basic-networking.md

```diff
@@ -0,0 +1,237 @@
+* The Backend container is a Spring Boot application that connects to the MySQL DB container and provides a REST service to the outside world.
+* The Frontend container is an AngularJS application that consumes the REST service from the backend container.
+
+You can find all the code examples on [Github](https://github.com/basmoorkens/docker-networking-demo){:target="_blank"}.
+The only thing needed to complete this guide is a working Docker installation.
+
+**Ready? Set. Go!**
+
+# Network Setup
+To start this guide, let's have a look at the Docker networks that are available on our machine:
+
+`docker network ls`  
+
+> When you run a Docker container and you do not provide any network settings, it is by default attached to the bridge network.
+
+Containers that are connected to this default bridge network can communicate with each other by using their internal Docker IP address.
+
+Docker does not support automatic service discovery on this network.
+
+We want to be able to access our containers by using their container name instead of the internal IP address so we are going to create our own network:
+
+`docker network create --driver bridge basic-bridge`  
+
+This creates a user defined network with the bridge driver that is called `basic-bridge`.
+If we look at the Docker network stack, we see that our user defined bridge network is added:
+
+<p>
+    <img class="image" alt="Network stack after adding basic-bridge" src="/img/docker-basic-networking/docker-network-ls-basic-bridge.png" />
+</p>
+
+You can look at more details of the network by using following command
+
+`docker network inspect basic-bridge`
+
+<p>
+    <img class="image" alt="Basic bridge details" src="/img/docker-basic-networking/bridged-network-details.png" />
+</p>
+
+The basic-bridge network can use IP addresses from the `172.18.0.0/16` range and will use `172.18.0.1` as its default gateway.
+
+# Database container
+
+Now that we have created our own network, we can start attaching our containers to that network.
+Let's start off by creating the MySQL database container.
+The following command pulls the MySQL image from the Docker repository and starts it as a container that is attached to our network:
+
+`docker run --name=mydb --network=basic-bridge -p 3306:3306 -e MySQL_ROOT_PASSWORD=test -d MySQL:8.0.3`
+
+That's easy right?
+This container is attached to the `basic-bridge` network that we created in the previous step.
+Run the following command to look at the container in detail:
+
+`docker inspect mydb`
+
+<p>
+    <img class="image" alt="docker inspect mydb" src="/img/docker-basic-networking/mydb-inspect.png" />
+</p>
+
+We can see in the output that it has gotten the `172.18.0.2` IP address and that it's using the default gateway of the network that we created.
+Now we should set up the database in our container with the schema for our REST application.
+
+`MySQL -h 127.0.0.1 -P 3306  --user=root --password=test`
```

Comment:
Is't commando niet `mysql` (all lowercase)?

Reply:
klopt. ik weet niet vanwaar die bizare formatting komt